---
title: "PCA_Analysis_lyrataData"
author: "Carina Bravo-Chan"
date: "2023-04-12"
output: pdf_document
---
```{r setup,warning=FALSE,include=FALSE}
knitr::opts_knit$set(echo=TRUE, root.dir='~/Desktop/BIO_400X_R/UNCG_DataWrangling_NewOld')
```

**Proposed work-flow for independent project:**

This data set is used on the ongoing research project in Dr. Remington's lab. In our lab we study plants' evolutionary life history and the molecular basis of phenotype divergence within populations. The system studied consists of two sub-populations of the rock cress *Arabdopsis lyrata*. One group from Spiterstulen (SP), Norway, and the other from Mayodan (MA), North Carolina. These populations were chosen because they occupy environments on opposite ends of the spectrum, and display different patterns of resource allocation for reproductive and vegetative growth. 
The data consists of phenotype records of traits from F2 individuals of *A. lyrata*, accompanied by the genotype of QTL regions. The hypothesis is that the resulting F2 phenotype is influenced by the type of allele inherited, Mayodan or Spiterstulen.
Individual observations were classified according to the number of allele copies, inherited from Mayodan parents, present in the genes of a QTL region. Individuals with **no** MA alleles, were identified with the number 0. It was expected plants with a more perennial-like phenotype, displaying a longer season of vegetative growth, and a short reproductive season with modest flowering. The heterozygous individuals were identified with the number 1 (one parental copy of the MA allele), and an intermediary phenotype was expected with resource allocation displaying displaying a mixed, or inconsistent, pattern. Finally, the individuals that were homozygous MA were identified with the number 2 (two parental copies of the MA allele). The expected phenotype was more annual-like, displaying stagnation of vegetative growth for overwintering, followed by an intense resource allocation for reproductive growth with significantly more bolting, branching, and flowering. 
The goal is to write a script to analyze the trends within this data set to find if there is any significant correlation between perennial traits and the underlying alleles. For example, a lower number of inflorescences suggesting a more perennial phenotype, and would allow us to examine if the genotype scoring for this, and other, traits would corroborate our hypothesis.


```{r setup,warning=FALSE,include=FALSE}
    knitr::opts_knit$set(root.dir='~/Desktop/BIO_400X_R/UNCG_DataWrangling_Old')
```

```{r, include=FALSE ,message=FALSE, warning=FALSE}
library(tidyverse)

# Read in the data
mtDataAll <- read.csv("QTLStudyTraitAnalysis_survMarkers.csv")

# Generate calculated fields:
mtDataAll <- mtDataAll %>%
  mutate(dDiam21 = Diam2 - Diam1,
         dDiam32 = Diam3 - Diam2,
         meanBLvs = rowMeans(select(., 12:14), na.rm = TRUE),
         meanOrder = rowMeans(select(., c(15, 18, 21)), na.rm = TRUE),
         meanBrNode = rowMeans(select(., c(16, 19, 22)), na.rm = TRUE),
         meanFlNode = rowMeans(select(., c(17, 20, 23)), na.rm = TRUE))

# Check variable names
colnames(mtDataAll)

# Check variable classes
mtDataAll %>%
  select(Pop, Fam, PShootStatus0215, LShootStatus0215) %>%
  map(class)

# Convert variables to factors
mtDataAll <- mtDataAll %>%
  mutate(across(c(Pop, Fam, PShootStatus0215, LShootStatus0215), as.factor))

# Create a new dataframe with just the F2 plants
mtDataF2 <- mtDataAll %>%
  filter(Pop == "F2")

# Check variable names
names(mtDataF2)
```

```{r, include=FALSE ,message=FALSE, warning=FALSE}
library(tidyverse)
# Decompose GI genotype into additive and dominance components
mtDataF2 <- mtDataF2 %>% 
  mutate(GIAdd = GI - 1,
         GIDom = 1 - abs(1 - GI))

# Calculate mean number of inflorescence branches
mtDataF2 <- mtDataF2 %>% mutate(meanBrNumber = meanFlNode - meanBrNode)

# Single-trait regressions on GI genotype
single_trait_models <- c("Diam3", "dDiam32", "Infl", "Diam2", "Diam1", "dDiam21", 
                         "meanBLvs", "meanOrder", "LatRating0215", "Rhiz0615", 
                         "meanFlNode", "meanBrNode", "meanBrNumber")

for (trait in single_trait_models) {
  formula <- formula(paste(trait, "~ GI"))
  model <- lm(formula, data = mtDataF2)
  print(summary(model))
  ggplot(mtDataF2, aes(x = GI, y = !!sym(trait))) +
    geom_point() +
    geom_smooth(method = "lm", se = FALSE, color = "blue") +
    labs(x = "GI", y = trait)
}

# Regressions on GI additive and dominance components
multi_trait_models <- c("Diam1", "Diam2")

for (trait in multi_trait_models) {
  formula <- formula(paste(trait, "~ GIAdd + GIDom"))
  model <- lm(formula, data = mtDataF2)
  print(summary(model))
  ggplot(mtDataF2, aes(x = GIAdd, y = !!sym(trait), color = GIDom)) +
    geom_point() +
    geom_smooth(method = "lm", se = FALSE) +
    labs(x = "GIAdd", y = trait, color = "GIDom")
}

# PC Analysis

PCTraitsF2dExt <- mtDataF2[,c(5,27,31,32,24,41,30,39,40,37)]
PCTraitsF2dExtTrim <- na.omit(PCTraitsF2dExt)
PCTraitsF2dExtImputed <- impute(PCTraitsF2dExt) # created a new data frame, using the impute function to force PCA despite empty rows within data frame

length(PCTraitsF2dExtImputed[,1])

names(PCTraitsF2dExtImputed)
#replace all is.na=0

PCModelF2dExt <- prcomp(PCTraitsF2dExtImputed, scale.=TRUE)
summary(PCModelF2dExt)
PCModelF2dExt$rotation

PCPredictF2 <- predict(PCModelF2dExt, newdata = mtDataF2)
mtDataF2 <- cbind(mtDataF2, PCPredictF2)

names(mtDataF2)

PC1_PIN <- lm(PC1 ~ PIN_combined, data = mtDataF2)
summary(PC1_PIN)

PC1_SOC1 <- lm(PC1 ~ SOC1, data = mtDataF2)
summary(PC1_SOC1)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
